distributor/distributor.go

```diff
@@ -452,14 +453,18 @@ func (d *Distributor) forAllIngesters(f func(cortex.IngesterClient) (interface{}
 	}
 
 	var lastErr error
-	result := []interface{}{}
+	result, numErrs := []interface{}{}, 0
 	for range ingesters {
 		select {
 		case resp := <-resps:
 			result = append(result, resp)
 		case lastErr = <-errs:
+			numErrs++
 		}
 	}
+	if numErrs > (d.cfg.ReplicationFactor - d.cfg.MinReadSuccesses) {
```

Comment:
I don't get it. Why do we need `numErrs`? Why not just track `len(result) > d.cfg.MinReadSuccesses`?